hw/syn/tools/dc/testsynth.tcl

```diff
@@ -0,0 +1,113 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Simple tcl script for DC to do some wire-load-model-based test syntheses.
+
+#####################
+##  PREPARE FLOW   ##
+#####################
+
+# tool setup
+source ../../../foundry/syn/dc/setup.tcl
+
+# paths
+set WORKLIB  "WORK"
+set REPDIR   "REPORTS"
+set DDCDIR   "DDC"
+set VLOGDIR  "NETLISTS"
+
+exec mkdir -p ${REPDIR} ${DDCDIR} ${VLOGDIR} ${WORKLIB}
+
+# define work lib path
+define_design_lib WORK -path $WORKLIB
+
+#######################
+##  DESIGN SOURCES  ###
+#######################
+
+set DUT "rv_plic_target"
+
+lappend search_path "../../../ip/prim/rtl/"
+set SRC {  "../../../ip/rv_plic/rtl/rv_plic_target.sv" }
+
+# additional defines
+set DEFINE ""
+
+# additional parameters
+set PARAMS "N_SOURCE=128"
+
+###########################
+##   ELABORATE DESIGN    ##
+###########################
+
+# delete previous designs.
+remove_design -designs
+sh rm -rf $WORKLIB/*
+
+analyze -define ${DEFINE} -format sv ${SRC}    > "${REPDIR}/${DUT}_analyze.rpt"
+elaborate  ${DUT} -parameters ${PARAMS}        > "${REPDIR}/${DUT}_elab.rpt"
+link                                           > "${REPDIR}/${DUT}_link.rpt"
+check_design                                   > "${REPDIR}/${DUT}_check.rpt"
+
+write_file -format ddc -hierarchy -output "${DDCDIR}/${DUT}_elab.ddc"
+write_file -format verilog -hierarchy -output "${DDCDIR}/${DUT}_elab.v"
+
+###########################
+##   APPLY CONSTRAINTS   ##
+###########################
+
+set CLK_PIN clk_i
+set RST_PIN rst_ni
+
+# timing constraint in ns
+# set timing to 250 MHz
+set TCK     4.0
+set IN_DEL  1.0
+set OUT_DEL 1.0
+set DELAY   ${TCK}
+
+create_clock ${CLK_PIN} -period ${TCK}
+
+set_ideal_network ${CLK_PIN}
+set_ideal_network ${RST_PIN}
+
+set_max_delay ${DELAY} -from [all_inputs] -to [all_outputs]
+set_input_delay ${IN_DEL} [remove_from_collection [all_inputs] {${CLK_PIN}}] -clock ${CLK_PIN}
+set_output_delay ${OUT_DEL}  [all_outputs] -clock ${CLK_PIN}
+
+set_driving_cell  -no_design_rule -lib_cell ${driving_cell} -pin X [all_inputs]
+set_load [load_of ${load_lib}/${load_cell}/A] [all_outputs]
+
+# set a nonzero critical range to be able to spot the violating paths better
+# in the report
+# set_critical_range 0.2 ${DUT}
+
+######################
+##    MAP DESIGN    ##
+######################
+
+compile_ultra -gate_clock -scan > "${REPDIR}/${DUT}_compile.rpt"
+
+# preserve hierarchy for reports
+#compile_ultra -gate_clock -scan -no_autoungroup > "${REPDIR}/${DUT}_compile.rpt"
+
+#################
+##   REPORTS   ##
```

Comment:
Looking at this, I think you might want to move the entire report_* set of commands into a separate script like "generate_reports.tcl" so that you can incorporate it in many/every place you want to run reports and only have to update it once.

Reply:
that is a good idea. I think I will start to move these things out into a synthesis library and import them into all synthesis scripts that we currently have.